crates/nu-command/src/network/http/client.rs

```diff
@@ -34,10 +34,17 @@ pub fn http_client(allow_insecure: bool) -> ureq::Agent {
         .build()
         .expect("Failed to build network tls");
 
-    ureq::builder()
+    let mut agent_builder = ureq::builder()
         .user_agent("nushell")
-        .tls_connector(std::sync::Arc::new(tls))
-        .build()
+        .tls_connector(std::sync::Arc::new(tls));
+
+    if let Some(http_proxy) = retrieve_http_proxy_from_env() {
+        if let Ok(proxy) = ureq::Proxy::new(http_proxy) {
```

Comment:
I'm not particularly convinced of the `if let` approach here, as the user won't be notified if ureq is unable to parse the proxy. This could give them the sense that they're making HTTP calls through a proxy when they're not.

On the other hand, this error seems quite unlikely to happen; I couldn't find a value that wasn't `Ok` for ureq to parse, even white space.

Let me know what you think: is it best to silently discard the proxy if unable to parse the `HTTP_PROXY` value, or should we `expect` a successful parsing?